### PR TITLE
New commit to this branch

### PR DESCRIPTION
**Pull request by atifmmahmud**
Some of the code still needs refactoring and optimization. I also have an additional feature in mind that I want to add. Test suite needs to be developed. Will complete them within the next week.

### CODE REVIEW

**Review by jungkumseok — CHANGES_REQUESTED** [review]
Hi Atif, I see some anti-patterns in the commit, some specific to Angular and others more general:

1. Pollution in global namespace:
    - There is a bunch of variable declarations you added at the top, and this is strongly discouraged as they may conflict with other unknown/future globals. Plus this makes the logic hard to follow. Please move all the global variables and make them local to the actual routine that needs the variables. If you need something akin to a "global" object, then either use the `$rootScope` or create a service using the `.factory()` function. If you need to use some of the variables as constants, you can declare them as so, using `.constant()` function.

2. Avoid calling functions in the HTML in directives like ng-show, as this function will be called in every digest cycle. E.g. codes.html:24 ng-show="$view.showOpenButtons()". You can imagine this as $view.showOpenButtons() being called every 25~50 ms building up a delay and eventually leading to a freeze. If you want to do what you're doing in ng-show, create a boolean flag and update the flag.

Please refer to Angular docs if you're not so sure about good/bad practices:
https://docs.angularjs.org/api

**Comment by atifmmahmud** [comment]
Hi Kumseok,

Thanks for the feedback. I am working on refactorting now; I have moved
around certain variable declarations and am sending you a screenshot here.
Please let me know if this will work. I am also work on the ng-show now.


Thanks
-Atif
